main.tf

```diff
@@ -15,7 +35,7 @@ resource "helm_release" "istio-discovery" {
   namespace        = var.istio_release_namespace
   create_namespace = true
   values           = [
-    templatefile("${path.module}/values-istiod.yaml", {})
+    templatefile("${path.module}/values/values-istiod.yaml", {})
```

Comment:
you already have a folder named values, `values-*.yaml` is redundant. `/values/istiod.yaml` is expressive enough.

Reply:
removed values from file names